src/mapgen/mg_decoration.cpp

```diff
@@ -117,16 +108,39 @@ bool Decoration::canPlaceDecoration(MMVManip *vm, v3s16 p)
 		v3s16( 1, 1, -1)
 	};
 
+
 	// Check these 16 neighboring nodes for enough spawnby nodes
-	for (size_t i = 0; i != ARRLEN(dirs); i++) {
-		u32 index = vm->m_area.index(p + dirs[i]);
+	for (size_t i = 0; i != ARRLEN(dirsNormal); i++) {
+		u32 index = vm->m_area.index(p + dirsNormal[i]);
 		if (!vm->m_area.contains(index))
 			continue;
 
 		if (CONTAINS(c_spawnby, vm->m_data[index].getContent()))
 			nneighs++;
 	}
 
+	if (check_offset != 0) {
+		static const v3s16 dirsOffset[8] = {
```

Comment:
The bug is here: `static` arrays are initialized once in a lifetime - in this case it happens on the first decoration.
Solution: remove `static`

Also in this project, variable names should follow the snake_case convention: https://dev.minetest.net/Code_style_guidelines#Miscellaneous